node/src/components/linear_chain.rs

```diff
@@ -62,6 +66,67 @@ impl<I> LinearChainComponent<I> {
     }
 }
 
+fn outcomes_to_effects<REv, I>(
```

Comment:
How about making this a method on the `Outcomes` type, so that you could do `self.linear_chain_state.handle_something(...).into_effects(effect_builder)` in the `handle_*` methods, instead of `let outcomes = ...; outcomes_to_effects(...)`?

Reply:
Not a bad idea! Although, on the second thought, it means that my `into_effects` method on `Outcomes` would have to define which effects it allows mapping into:
```rust
impl Outcomes {
  fn into_effects<REv, I>(self, effect_builder: EffectBuilder<REv>) -> Effects<Event<I>>
  where
    REv: From<StorageRequest>
        + From<NetworkRequest<I, Message>>
        + From<LinearChainAnnouncement>
        + From<ContractRuntimeRequest>
        + From<ChainspecLoaderRequest>
        + Send,
    I: Display + Send + 'static,
{ … }
```
and I don't want to make the effect-unaware `Outcomes` to now know about that.

Reply:
I see. Alright, I don't think it's a blocker, anyway - so whichever approach suits you best :)